tests/test_git_fetch.rs

```diff
@@ -0,0 +1,368 @@
+    @ 43fe77b5815e descr_for_b b
+    | o cb1fe5d97488 descr_for_a2 a2
+    |/  
+    | o 5f35ea16f836 descr_for_a1 a1
+    |/  
+    o 9929b494c411 trunk1 master trunk1
+    o 000000000000 (no description set) 
+    "###);
+
+    // Nothing in our repo before the fetch
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    @ 230dd059e1b0 (no description set) 
+    o 000000000000 (no description set) 
+    "###);
+    let stdout = test_env.jj_cmd_success(&target_jj_repo_path, &["git", "fetch", "--glob", "b"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    o 43fe77b5815e descr_for_b b
+    o 9929b494c411 trunk1 
+    | @ 230dd059e1b0 (no description set) 
+    |/  
+    o 000000000000 (no description set) 
+    "###);
+    let stdout = test_env.jj_cmd_success(&target_jj_repo_path, &["git", "fetch", "--glob", "a*"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    o cb1fe5d97488 descr_for_a2 a2
+    | o 5f35ea16f836 descr_for_a1 a1
+    |/  
+    | o 43fe77b5815e descr_for_b b
+    |/  
+    o 9929b494c411 trunk1 
+    | @ 230dd059e1b0 (no description set) 
+    |/  
+    o 000000000000 (no description set) 
+    "###);
+    let stdout = test_env.jj_cmd_success(&target_jj_repo_path, &["git", "fetch", "--glob", "a1"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Nothing changed.
+    "###);
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    o cb1fe5d97488 descr_for_a2 a2
+    | o 5f35ea16f836 descr_for_a1 a1
+    |/  
+    | o 43fe77b5815e descr_for_b b
+    |/  
+    o 9929b494c411 trunk1 
+    | @ 230dd059e1b0 (no description set) 
+    |/  
+    o 000000000000 (no description set) 
+    "###);
+
+    // Change the target repo
+    let source_log = create_trunk2_and_rebase_branches(&test_env, &source_git_repo_path);
+    insta::assert_snapshot!(source_log, @r###"
+    Source git repo contents:
+    o f561acfed1d2 descr_for_b b
+    | o bae45043dbd2 descr_for_a2 a2
+    |/  
+    | o 51fa81c130b6 descr_for_a1 a1
+    |/  
+    @ c0bc3b78f807 descr_for_trunk2 trunk2
+    o 9929b494c411 trunk1 master trunk1
+    o 000000000000 (no description set) 
+    "###);
+
+    // Our repo before and after fetch
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    o cb1fe5d97488 descr_for_a2 a2
+    | o 5f35ea16f836 descr_for_a1 a1
+    |/  
+    | o 43fe77b5815e descr_for_b b
+    |/  
+    o 9929b494c411 trunk1 
+    | @ 230dd059e1b0 (no description set) 
+    |/  
+    o 000000000000 (no description set) 
+    "###);
+    let stdout = test_env.jj_cmd_success(
+        &target_jj_repo_path,
+        &["git", "fetch", "--glob", "b", "--glob", "a1"],
+    );
+    insta::assert_snapshot!(stdout, @"");
+    // TODO: Is it a bug that the old commits are not abandoned?
```

Comment:
Is this a bug?

Reply:
I haven't thought much about it, but it sounds like it might be https://github.com/martinvonz/jj/issues/864

Reply:
The target repo isn't colocated though.

Reply:
I marked this as a BUG, will create the corresponding issue when I merge this. I still don't know what's going on.